### DISCUSSION TITLE
Making a fork from node

### DISCUSSION THREAD

**Discussion by sulaiman21** [xyflow/xyflow#3338] [Q&A]
Hi all  🙋‍♂️!

I am using reactflow to make some customization according our needs. But i wanted to add an edge that connects 2 nodes to its parent node. For better understanding please see the below image.

https://user-images.githubusercontent.com/58777040/261019695-a02dd63c-9313-42be-ad85-c8a586030928.png

Currently i tried to add a node with just handles i.e source handle and target handle, without any label etc.
Which is looking like

https://user-images.githubusercontent.com/58777040/261020088-2c37b5e3-13b4-43d0-846a-fd15a42b8d2d.png

But the layout is really bad, I am using dagre for auto layout at the moment.

Any help and suggestion would be appreciated.

thanks.

**Comment by moklick** [comment]
Hey @sulaiman21 

what is the issue / question here? Can you share a codesandbox?

**Reply by sulaiman21** [reply]
Well i am trying to implement a fork feature where user can click on a button to extend the node and split it into 2 nodes, which i have already done but the issue is coming with auto layout. It doesn't do the layout now.
